pf4j/src/main/java/org/pf4j/PropertiesPluginDescriptorFinder.java

```diff
@@ -68,6 +68,13 @@ protected Properties readProperties(Path pluginPath) throws PluginException {
 
         log.debug("Lookup plugin descriptor in '{}'", propertiesPath);
         if (Files.notExists(propertiesPath)) {
+
+            try {
+                propertiesPath.getFileSystem().close();
```

Comment:
I am not so confident in this line.
The idea is that you don't know the type of FileSystem and probably this line doesn't work in all scenario.
Probably your case is to find `plugin.properties` in a jar/zip. It's another usual scenario (it's used always by me in my project) to run the application together with plugins from IDEA IntelliJ. In this case the PF4J try to find `plugin.properties` of each plugin in the root of each plugin:
![image](https://user-images.githubusercontent.com/1248311/66373452-5a997380-e9b1-11e9-8331-6051096e9e52.png)
In this case probably the message of exception will be `UnsuportedOperationException`.
If we want to complicate a little bit the things, we have Windows's FileSystem and Linux/Unix's File System.
So, this modification requires more tests.

Reply:
Agree. I only tested the case with jar files.